File: Java/1160. Find Words That Can Be Formed by Characters.java
class Solution {
    public int countCharacters(String[] words, String chars) {
        int ans = 0, flag;
        HashMap<Character, Integer> dict = new HashMap<>(), temp = new HashMap<>();

        for(char i: chars.toCharArray()) dict.put(i, (dict.get(i) == null ? 1 : dict.get(i) + 1));

        for(String i: words){
            temp.putAll(dict);
            flag = 1;

            for(char j: i.toCharArray()){
                if(temp.get(j) == null || temp.get(j) == 0){
                    flag = 0;
                    break;
                }
                temp.put(j, (temp.get(j) - 1));
            }

            if(flag == 1) ans += i.length(); 
        }

        return ans;
    }
}

File: Python/1160. Find Words That Can Be Formed by Characters.py
class Solution:
    def countCharacters(self, words: List[str], chars: str) -> int:
        res = 0

        for i in words:
            t = copy.deepcopy(chars)
            flag = 1

            for j in i:
                if j not in t:
                    flag = 0
                    break
                t = t.replace(j, "", 1)
            
            if flag: res += len(i)
        
        return res
